### PR TITLE
sidebar reorganization

### PR DESCRIPTION
**Pull request by pdg744**
This PR mostly just moves content around in order to make the conceptual flow clearer. 
Rather than looking at the diff in GitHub, I suggest reviewing by opening the deployment, imagining yourself as a new user, and trying it out. 

The high-level thinking around the conceptual flow for users is described in this diagram: 
image

- [ ] This PR moves the `bonsai/quickstart.md` doc to `generating-proofs/remote-proving.md`. If we use this organization, we need to set up a re-direct so that the QR codes and outstanding links that point to `bonsai/quickstart` don't 404. 
- [ ] This PR deletes the `bonsai/overview.md` doc. We need to set up a re-direct so that QR codes and outstanding links that point to `bonsai/` don't 404. The redirect should point to `generating-proofs/remote-proving.md` 

<s>This PR also drops the slug on the `zkvm_overview` doc, which means links to `/zkvm/` will 404. If we want to drop this slug, we should set up a re-direct from `/zkvm/` to `/zkvm/zkvm_overview.md` to address this. </s> resolved by keeping the slug

### CODE REVIEW

File: website/api/generating-proofs/dev-mode.md
Comment (flaub) [inline website/api/generating-proofs/dev-mode.md:4]: This shouldn't be needed

File: website/api/use-cases.md
Comment (flaub) [inline website/api/use-cases.md:61]: Don't use `/api/next`, this makes it hard to snap a new release. Use relative paths, see: https://docusaurus.io/docs/versioning#link-docs-by-file-paths

File: website/api/use-cases.md
Comment (flaub) [inline website/api/use-cases.md:57]:
```suggestion
1. [Write an application for the RISC Zero zkVM][zkvm-quickstart].
```

File: website/api/use-cases.md
Comment (flaub) [inline website/api/use-cases.md:61]:
```suggestion
[zkvm-quickstart]: ./zkvm/quickstart.md
```

File: website/api/use-cases.md
Comment (flaub) [inline website/api/use-cases.md:58]:
```suggestion
2. [Generate proofs for your zkVM application][bonsai-quickstart].
```

File: website/api/use-cases.md
Comment (flaub) [inline website/api/use-cases.md:62]:
```suggestion
[bonsai-quickstart]: ./bonsai/quickstart.md
```

File: website/api/use-cases.md
Comment (flaub) [inline website/api/use-cases.md:51]: Maybe this sort of statement belongs on the marketing website, but it doesn't feel right for devloper docs.

File: website/api/getting-started.md
Comment (nategraf) [inline website/api/getting-started.md:60]: `./bonsai/quickstart` appears to no longer be in the sidebar and is kind of stranded. it seems the remote proving page has all the information that used to be there, so linking there seems like the way to go, or simply to the "proving options" page. Either way, we should probably delete the Bonsai quickstart page.

**Review by nategraf — COMMENTED** [review]
This looks quite good. Definitely a marked improvement over the current documentation structure.

Note that this will appear under "next" version, although all of it seems relevant to `0.20`. Once this sis essentially ready, it would be good to copy the changes there so folks land on the updated docs. (That or we simply release `0.21` very soon)

**Review by Cardosaum — COMMENTED** [review]
The content is great and love the reorganization!

Made some comments but I think we could land this and iterate on a future PR if appropriate.

**Comment by nategraf** [comment]
Additional note that once we are happy with the new structure, we should add redirects before merging this PR.

File: website/api/generating-proofs/local-proving.md
Comment (flaub) [inline website/api/generating-proofs/local-proving.md:3]: Can we drop all the slugs?

File: website/api/generating-proofs/remote-proving.md
Comment (flaub) [inline website/api/generating-proofs/remote-proving.md:37]: Duplicate link

File: website/api/getting-started.md
Comment (pdg744) [reply website/api/getting-started.md:57]: I'll defer to you and/or Angelo on that question.

Comment (pdg744) [reply website/api/getting-started.md]: i've updated to point to bonsai-on-eth for now.

File: website/api/generating-proofs/remote-proving.md
Comment (pdg744) [reply website/api/generating-proofs/remote-proving.md:17]: fixed

File: website/api/use-cases.md
Comment (pdg744) [reply website/api/use-cases.md:10]: good call-out!  

if you have a suggested revision i'm happy to include an improvement to this on this PR, otherwise let's save it for a future improvement -- I'm just copying/pasting this section from elsewhere on the site.

File: website/api/use-cases.md
Comment (pdg744) [reply website/api/use-cases.md:24]: good call!

File: website/api/zkvm/zkvm_overview.md
Comment (pdg744) [reply website/api/zkvm/zkvm_overview.md:4]: feel free to open an issue for this; i'm not going to worry about it on this PR

File: website/api/zkvm/zkvm_overview.md
Comment (pdg744) [reply website/api/zkvm/zkvm_overview.md:47]: i'm not sure what the best source of truth is for feature flags info

File: website/api/getting-started.md
Comment (pdg744) [reply website/api/getting-started.md]: interesting option to embed content from one page in another! i've revised to avoid the redundancy, but mb we want to take advantage of that embed option...

File: website/api/getting-started.md
Comment (pdg744) [reply website/api/getting-started.md:1]: ya true. revised!

File: website/api/use-cases.md
Comment (nategraf) [reply website/api/use-cases.md:10]: Ya, I dislike the way this section is written. I wrote some suggestions, although I'm still not totally very happy with it https://github.com/risc0/risc0/pull/1484

Comment (pdg744) [reply website/api/use-cases.md]: awesome -- thanks victor!

**Comment by pdg744** [comment]
thanks for the feedback, y'all! i think i've addressed all the feedback that it makes to address on this PR. 
if there's no further feedback, i'll connect with morgan to set up the appropriate re-directs and we can get this landed.

**Comment by capossele** [comment]
This is a great improvement! I have few minor feedbacks that do not necessarily need to be considered within this PR.
- I feel there is a gap between the getting started and the actual information the user needs to get started without getting distracted by secondary details. I'm mostly referring to the **What is a zkVM application** section. In there we have some *101s* that I personally find not that easy to follow unless knowing already a basic end to end flow. 
- In section **Writing your zkVM Application** we have a bunch of different type of resources; I'm not convinced about having the zkVM spec in this section. Also I feel the *rust resources* should be moved after the tutorial.

**Comment by nategraf** [comment]
> thanks for the feedback, y'all! i think i've addressed all the feedback that it makes to address on this PR. if there's no further feedback, i'll connect with morgan to set up the appropriate re-directs and we can get this landed.
> 
> feel free to add more comments if there are other things that we should improve here.

I'm happy with these changes, and think this is a marked improvement. I'd be happy to see further content work happen in follow-up PRs.

**Comment by pdg744** [comment]
OK the only remaining todo here is to make sure we're not killing links from devrel/marketing assets. I think we can manage the redirects via the docusaurus.config.js file, but I struggled to figure it out when I tried earlier. If someone's available to pair with me for 30 minutes, I think we can land this! 

(Here's a list of links we need to be careful about: https://docs.google.com/spreadsheets/d/1U3CAIzdsuX4eQQIfof9wkCXt0LYRjF8hp9KeK8uPVyY/edit#gid=0)

File: website/api/zkvm/zkvm_overview.md
Comment (flaub) [reply website/api/zkvm/zkvm_overview.md:47]: I think currently it's the README.

File: website/api/zkvm/zkvm_overview.md
Comment (flaub) [inline website/api/zkvm/zkvm_overview.md:4]: I think we need to keep this one

File: website/api/introduction.md
Comment (flaub) [inline website/api/introduction.md:4]: I think we need to keep this one too

**Review by flaub — APPROVED** [review]

**Pull request merged**
